src/ui/Search/searchinfilesworker.cpp

```diff
@@ -22,53 +22,43 @@ SearchInFilesWorker::~SearchInFilesWorker()
 
 void SearchInFilesWorker::run()
 {
-    // Search string converted to a regex
-    QString rawSearch = frmSearchReplace::rawSearchString(m_string, m_searchMode, m_searchOptions);
+    FileSearchResult::SearchResult searchResult;
+    searchResult.search = m_string;
 
     QFlags<QRegularExpression::PatternOption> options = QRegularExpression::NoPatternOption;
-    if (m_searchOptions.MatchCase == false) {
-        options |= QRegularExpression::CaseInsensitiveOption;
-    }
+    QFlags<QDirIterator::IteratorFlag> dirIteratorOptions = QDirIterator::NoIteratorFlags;
 
-    QRegularExpression regex(rawSearch, options);
+    if (m_searchMode == SearchHelpers::SearchMode::Regex) {
+        if (m_searchOptions.MatchCase == false) {
+            options |= QRegularExpression::CaseInsensitiveOption;
+        }
+        QString rawSearch = frmSearchReplace::rawSearchString(m_string, m_searchMode, m_searchOptions);
+        m_regex.setPattern(rawSearch);
+        m_regex.setPatternOptions(options);
+    }else if (m_searchMode == SearchHelpers::SearchMode::SpecialChars) {
+        QString rawSearch = frmSearchReplace::rawSearchString(m_string, m_searchMode, m_searchOptions);
+        m_string = unescapeString(m_string);
```

Comment:
Barring any other issues. I think this is finally ready to be merged.

I learned a lot of nifty tricks during this little project.  Our search 
functions are actually faster than kate's now from testing. :)

Also,  I did an experimental test of threading the searches just to 
check.  In cases of SSD hard drives, it actually increases the 
performance even further.  It would be a nice little feature to have as 
optional later down the line, but this should be plenty fast for now.